### PR TITLE
Use new version of provisio plugin

### PR DESCRIPTION
**Pull request by electrum**
This includes a new version of the Takari Archiver which supports
hardlinks in archives.

### CODE REVIEW

File: presto-server/src/main/provisio/presto.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <runtime>
     <!-- Target -->
-    <archive name="${project.artifactId}-${project.version}.tar.gz" />
+    <archive name="${project.artifactId}-${project.version}.tar.gz" hardLinkIncludes="**/*.jar" />
```
Comment (findepi) [inline presto-server/src/main/provisio/presto.xml]: How does the produced .tar.gz work on a file system that doesn't support hardlinks?
How does it work on Windows?

File: pom.xml [hunk 2]
```diff
@@ -1304,14 +1304,14 @@
             <plugin>
                 <groupId>io.prestosql</groupId>
                 <artifactId>presto-maven-plugin</artifactId>
-                <version>6</version>
+                <version>8</version>
```
Comment (findepi) [inline pom.xml]: Per previous discussion (https://github.com/prestosql/presto/pull/1690/files#r332371929)

I am OK based on @jvanzyl explanation that this is not a requirement for provisio upgrade for server tarball.
Please extract to separate commit (maybe as a latter commit).

File: presto-server/src/main/provisio/presto.xml (hunk 1 above)
Comment (jvanzyl) [reply presto-server/src/main/provisio/presto.xml]: I've literally used Windows five times in my entire life, so someone would need to try it but I assume they would use the `--hard-dereference` option or equivalent. The tar implementations on Windows might even dereference hardlinks given they aren't supported.

File: pom.xml (hunk 2 above)
Comment (electrum) [reply pom.xml]: Done

File: .travis.yml [hunk 3]
```diff
@@ -61,6 +61,7 @@ git:
   quiet: true
 
 install:
+  - rm -f ~/.m2/settings.xml
```
Comment (findepi) [inline .travis.yml]: uh... why? Please leave some explanation in code comment

Comment (kokosing) [reply .travis.yml]: what is set there by default?

**Comment by jvanzyl** [comment]
The RPM, using the tarball with hardlinks, is also reduced in size.

File: .travis.yml (hunk 3 above)
Comment (electrum) [reply .travis.yml]: Sorry, was a hack to allow the build to run, not intended to be permanent. Travis uses the Google mirror of Central, but the artifact hadn't synced yet after over 24 hours. I filed an issue with Sonatype to investigate.

This is what their default config file looks like:
```xml
<settings>
  <profiles>
    <profile>
      <id>standard-with-extra-repos</id>
      <activation>
        <activeByDefault>true</activeByDefault>
      </activation>
      <repositories>
        <repository>
          <id>central</id>
          <name>Central Repository</name>
          <url>https://repo.maven.apache.org/maven2</url>
          <releases>
            <enabled>true</enabled>
          </releases>
          <snapshots>
            <enabled>false</enabled>
          </snapshots>
        </repository>
        <repository>
          <id>sonatype</id>
          <name>OSS Sonatype repo (releases)</name>
          <releases>
            <enabled>true</enabled>
            <updatePolicy>always</updatePolicy>
            <checksumPolicy>warn</checksumPolicy>
          </releases>
          <snapshots>
            <enabled>false</enabled>
            <updatePolicy>never</updatePolicy>
            <checksumPolicy>fail</checksumPolicy>
          </snapshots>
          <url>https://oss.sonatype.org/content/repositories/releases/</url>
        </repository>
        <repository>
          <id>sonatype-snapshots</id>
          <name>OSS Sonatype repo (snapshots)</name>
          <releases>
            <enabled>false</enabled>
            <updatePolicy>always</updatePolicy>
            <checksumPolicy>warn</checksumPolicy>
          </releases>
          <snapshots>
            <enabled>true</enabled>
            <updatePolicy>never</updatePolicy>
            <checksumPolicy>fail</checksumPolicy>
          </snapshots>
          <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
        </repository>
        <repository>
          <id>sonatype-apache</id>
          <name>Apache repo (releases)</name>
          <releases>
            <enabled>true</enabled>
            <updatePolicy>always</updatePolicy>
            <checksumPolicy>warn</checksumPolicy>
          </releases>
          <snapshots>
            <enabled>false</enabled>
            <updatePolicy>never</updatePolicy>
            <checksumPolicy>fail</checksumPolicy>
          </snapshots>
          <url>https://repository.apache.org/content/repositories/releases/</url>
        </repository>
        <repository>
          <id>apache-snapshots</id>
          <name>ASF repo (snapshots)</name>
          <releases>
            <enabled>false</enabled>
            <updatePolicy>never</updatePolicy>
            <checksumPolicy>warn</checksumPolicy>
          </releases>
          <snapshots>
            <enabled>true</enabled>
            <updatePolicy>always</updatePolicy>
            <checksumPolicy>fail</checksumPolicy>
          </snapshots>
          <url>https://repository.apache.org/snapshots/</url>
        </repository>
      </repositories>
    </profile>
  </profiles>
<mirrors>
    <mirror>
      <id>google-maven-central</id>
      <name>GCS Maven Central mirror</name>
      <url>https://maven-central.storage-download.googleapis.com/repos/central/data/</url>
      <mirrorOf>central</mirrorOf>
    </mirror>
  </mirrors></settings>
```

**Comment by electrum** [comment]
The user experience is that the tarball, RPM, and on disk size are reduced in size (hardlinks are preserved when unpacking).
